src/components/tl/nccl/tl_nccl_team.c

```diff
@@ -0,0 +1,174 @@
+{
+    ucc_tl_nccl_context_t *ctx    =
+        ucc_derived_of(tl_context, ucc_tl_nccl_context_t);
+    ucc_status_t status;
+
+    UCC_CLASS_CALL_SUPER_INIT(ucc_tl_team_t, &ctx->super);
+    self->oob       = params->params.oob;
+    self->size      = self->oob.participants;
+    self->rank      = params->rank;
+    self->unique_id = ucc_malloc(sizeof(ncclUniqueId) * (self->size + 1),
+                                 "tl_nccl_unique_id");
+    if (!self->unique_id) {
+        tl_error(ctx->super.super.lib,
+                 "failed to allocate %zd bytes for unique_id array",
+                 sizeof(ncclUniqueId) * (self->size + 1));
+        return UCC_ERR_NO_MEMORY;
+    }
+    if (self->rank == 0) {
+        NCCLCHECK_GOTO(ncclGetUniqueId(&self->unique_id[self->size]),
+                       free_unique_id, status, ctx->super.super.lib);
+    }
+    status = self->oob.allgather(&self->unique_id[self->size], self->unique_id,
+                                 sizeof(ncclUniqueId), self->oob.coll_info,
+                                 &self->oob_req);
+    if (status != UCC_OK) {
+        tl_error(ctx->super.super.lib, "failed to start oob allgather");
+        goto free_unique_id;
+    }
+    return UCC_OK;
+
+free_unique_id:
+    ucc_free(self->unique_id);
+    return status;
+}
+
+UCC_CLASS_CLEANUP_FUNC(ucc_tl_nccl_team_t)
+{
+    tl_info(self->super.super.context->lib, "finalizing tl team: %p", self);
+    if (self->nccl_comm) {
+        ncclCommDestroy(self->nccl_comm);
+        cudaStreamDestroy(self->stream);
+    }
+}
+
+UCC_CLASS_DEFINE_DELETE_FUNC(ucc_tl_nccl_team_t, ucc_base_team_t);
+UCC_CLASS_DEFINE(ucc_tl_nccl_team_t, ucc_tl_team_t);
+
+ucc_status_t ucc_tl_nccl_team_destroy(ucc_base_team_t *tl_team)
+{
+    // ucc_tl_nccl_team_t *team = ucc_derived_of(tl_team, ucc_tl_nccl_team_t);
+    UCC_CLASS_DELETE_FUNC_NAME(ucc_tl_nccl_team_t)(tl_team);
+    return UCC_OK;
+}
+
+ucc_status_t ucc_tl_nccl_team_create_test(ucc_base_team_t *tl_team)
+{
+    ucc_tl_nccl_team_t *team = ucc_derived_of(tl_team, ucc_tl_nccl_team_t);
+    ucc_status_t status;
+    ncclResult_t nccl_status;
+
+    status = team->oob.req_test(team->oob_req);
+    if (status < 0) {
+        team->oob.req_free(team->oob_req);
+        tl_error(team->super.super.context->lib, "oob req test failed");
+        goto free_unique_id;
+    }
+    if (status == UCC_INPROGRESS) {
+        return UCC_INPROGRESS;
+    }
+    status = team->oob.req_free(team->oob_req);
+    if (status != UCC_OK) {
+        tl_error(team->super.super.context->lib, "oob req free failed");
+        goto free_unique_id;
+    }
+    CUDACHECK_GOTO(cudaStreamCreateWithFlags(&team->stream,
+                   cudaStreamNonBlocking), free_unique_id, status,
+                   team->super.super.context->lib);
+    nccl_status = ncclCommInitRank(&team->nccl_comm,team->size,
+                                   team->unique_id[0], team->rank);
+    if (nccl_status != ncclSuccess) {
```

Comment:
NCCLCHECK_GOTO ?

Reply:
I don't use NCCLCHECK here because it will print error. Instead in case of failure we print info message and continue.
We can consider additional paramer that tells TL if context create or team create is fatal error or not